frontend/src/lib/components/widgets/NumberInput/NumberInput.tsx

```diff
@@ -356,15 +396,42 @@ class NumberInput extends React.PureComponent<Props, State> {
           <UIInput
             type="number"
             inputRef={this.inputRef}
-            value={formattedValue}
+            value={
```

Comment:
nit: `value={this.state.value ?? formattedValue}`

Reply:
Updated 👍